Port to React-Native >= 0.7x and lottie-react-native@6.7.2

attarchi opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new React Native example application with splash screen functionality using Lottie animations.
	- Added support for different React Native versions and updated installation instructions.
  
- **Documentation**
	- Enhanced the README.md with detailed setup instructions, version compatibilities, and troubleshooting tips.

- **Bug Fixes**
	- Updated the splash screen management to ensure smoother transitions and animations.

- **Configuration**
	- Added new configuration files for Yarn, ESLint, Prettier, and Metro to improve development experience.

- **Tests**
	- Introduced unit tests for the new application to ensure component rendering and functionality.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

attarchi committed:
Port yarn to 3.6.4

attarchi committed:
Bump dependencies version

attarchi committed:
Convert android native codes to kotlin

attarchi committed:
Bump package version

attarchi committed:
Update readme

attarchi committed:
Integrating with lottie-ios@4.4.3

attarchi committed:
spelling of readme

attarchi committed:
Update yarn lock

erie-e9 commented on package.json:
```diff
@@ -36,11 +36,12 @@
-    "lottie-react-native": "^5.1.3"
-  }
-}
+    "lottie-ios": "4.4.3",
```
Why not use "lottie-ios": "4.5.0"? it's the latest version.

attarchi replied:
I want to thank you so much for your interest.
Because lottie-react-native uses this version.

https://github.com/lottie-react-native/lottie-react-native/blob/f230b98dbc943dc5c0c1fe61708bad8251198be9/packages/core/lottie-react-native.podspec#L25

erie-e9 replied:
oh right. Thanks for upgrade this package, I really love it ❤️

ashwini-ksingh commented:
@attarchi @erie-e9 When are we planning to release this?

attarchi commented:
I have no idea when the repo owner wants to do it but meanwhile, you can add this to your package.json.

```
    "react-native-lottie-splash-screen": "https://github.com/attarchi/react-native-lottie-splash-screen",
```

ashwini-ksingh commented:
@attarchi I did try using your fork, but getting [this](https://github.com/HwangTaehyun/react-native-lottie-splash-screen/issues/101) error with RN 0.75.4

attarchi commented:
> @attarchi I did try using your fork, but getting [this](https://github.com/HwangTaehyun/react-native-lottie-splash-screen/issues/101) error with RN 0.75.4

I have tested 0.75.4, and there is no error. Please remove the previous one and set it up from begin with this [instruction](https://github.com/attarchi/react-native-lottie-splash-screen/blob/master/README.md).

ashwini-ksingh commented:
@attarchi It was some issue with the build schemes configured in my project. Recreated the scheme and build was successful.

However, on ios, the lottie splash is immediately removed after load, and does not complete the animation. (tried with setAnimationFinished true/false, but its removed immediately). This is only happening in release build, works fine on debug.

attarchi commented:
@HwangTaehyun Cloud you please check this PR?

QaysTrad commented:
> I have no idea when the repo owner wants to do it but meanwhile, you can add this to your package.json.
> 
> ```
>     "react-native-lottie-splash-screen": "https://github.com/attarchi/react-native-lottie-splash-screen",
> ```

Thanks for updating the package,

I have installed the package from your repo, 
but when I use
RNRestart.restart()
the lottie splash it's run again on the iOS, until I close the app and reopen it 
and when I use LottieSplashScreen.show()
the app froze with no response

"react-native": "0.74.0",
"react": "18.2.0",
"react-native-lottie-splash-screen": "https://github.com/attarchi/react-native-lottie-splash-screen",

attarchi committed:
Move previous examples to a sub folder

attarchi committed:
Add react-native 0.7x CLI example project

attarchi committed:
Define yarn workspaces and define the project dependencies

attarchi committed:
Setup package in the react-native 0.7x example project

attarchi committed:
Fix react-native 0.76 building bug

attarchi committed:
Fix checksum error on example project

attarchi committed:
run iOS pod install for react-native 7x example project

attarchi committed:
add lottie-react-native package in example project for iOS pods

attarchi committed:
Setup package into the example 0.7x project for iOS

attarchi committed:
Add some comments

attarchi committed:
Remove the react show method

attarchi commented:
Hi @QaysTrad,

Thank you for your feedback and for testing the package!

I'll look into the issue with RNRestart.restart() causing the Lottie splash screen to reappear on iOS. There might be a conflict there that I'll need to resolve.

Regarding LottieSplashScreen.show(), I have actually removed that method as it didn't align with the package's purpose. This package is designed to handle the native splash screen while React Native is initializing. Once the app is fully loaded and you have access to React components, it's better to control Lottie animations directly using the lottie-react-native package. This package's focus is on the startup process, so handling splash screens beyond that isn't necessary.

Let me know if you need any further clarification!

attarchi committed:
Merge pull request #1 from attarchi/new-example

New example

attarchi commented:
Hi @ashwini-ksingh,

I've added an example project to the source code for running this app with RN 0.7x, and updated the package. Please check out the latest version.

Regarding your issue, keep in mind that when `setAnimationFinished` is set to `true`, the application won’t wait for the Lottie animation to finish before removing the splash screen. To ensure the splash screen waits for the animation to complete, set this value to **false** and clear the `build` folder inside your iOS project directory. 

Here’s how you can set it in your code:

```objc
// AppDelegate.mm file

    // Make the splash screen wait for the Lottie animation to finish
    [RNSplashScreen setAnimationFinished:false];
```
After making these changes, rebuild the project, and the splash screen should behave as expected in the release build. Let me know if this resolves the issue!

erie-e9 commented:
Hi again, I just followed instructions and all seem to be right, app is running without problems using this package v2, but I'm not sure if it 'cuz I'm using latest versions of all, I mean I'm testing new project with latest version of Xcode, metro, react@18.3.1, react-native@0.76.0, iOS simulator with iOS 18 and I notice it runs very quick, I can see my splash-screen only for 0.5 seconds and it only shows it when I close and reopen the app, nothing is displayed if I reload it in terminal.

Note: I know Lottie file is in the splash-screen because that little black square with white words in right side is part of it, it's like a watermark.
Note 2: I tested it with RNSplashScreen setAnimationFinished = false and true.

I just read the latest version of metro and react-native help to improve build time but I'd like to be sure that's the reason.

![Simulator Screen Recording - iPhone 16 Pro - 2024-10-25 at 15 55 18](https://github.com/user-attachments/assets/720e0ac2-fd05-4236-9115-caee7d77f1c6)

ashwini-ksingh commented:
@erie-e9 I am also facing the same behavior (though I am on RN 0.75.4)
Not sure why, its the same using the example in the repo as well (example/RN.0.7x-cli) @attarchi

singhayush1403 commented:
@erie-e9 Have the same issue. The lottie doesn't finish but the screen is removed

attarchi commented:
@singhayush1403
@ashwini-ksingh
@erie-e9

Thank you very much for your collaboration. I noticed the issue but couldn’t fix it, as I don’t have extensive experience with Objective-C and iOS app initialization. I’ll publish this version in my repo as is. Could you please open a separate issue to address the iOS problem in a future update? If anyone can help with this, feel free to jump in with a PR. Thanks!

attarchi commented:
I've published the latest changes to npm! You can check out the package here: [@attarchi/react-native-lottie-splash-screen on NPM](https://www.npmjs.com/package/@attarchi/react-native-lottie-splash-screen).

Feel free to install it and continue bug fixing.

```bash
$ yarn add @attarchi/react-native-lottie-splash-screen
```

attarchi committed:
Move images to the screenshot directory to decrease npm package size.

ashwini-ksingh commented:
@attarchi 

This seems to be an issue with the default zIndex given to App component, compared to the LottieAnimationView view added for splash.

I added `animationView.reactZIndex = 1;` in `createAnimationView` function, and it started working for me on 0.75.4 with new architecture and bridgeless enabled.

attarchi committed:
Fix the 7x iOS example layer zIndex

attarchi commented:
@ashwini-ksingh  
Thanks! Your solution worked perfectly.

Since the old architecture doesn’t support `reactZIndex`, I set `animationUIView.layer.zPosition = 1` in `AppDelegate.mm`, and it resolved the issue.

@erie-e9  
The example project has been updated and should work as expected. Also, you can also add `animationUIView.layer.zPosition = 1` directly to your `AppDelegate.mm` file.

erie-e9 commented:
Somebody else facing these errors on Android?

### EDIT:
Solution enable next ones in gradle.properties:
android.useAndroidX=true
android.enableJetifier=true

Steps:
- [ ] Init new RN project - @0.76.x.
- [ ] Install latest version of this dep - yarn add @attarchi/react-native-lottie-splash-screen@2.0.4
- [ ] Sync gradle on Android studio.
- [ ] Rebuild project: Build output will display next problem.

### Error 1
<img width="861" alt="Captura de pantalla 2024-10-31 a la(s) 5 05 46 p m" src="https://github.com/user-attachments/assets/5884cf45-1bde-4757-9aae-901c13295a24">

### Error 2
Unresolved reference Lottie in MainActivity.kt
<img width="877" alt="Captura de pantalla 2024-10-31 a la(s) 5 19 48 p m" src="https://github.com/user-attachments/assets/98a6cacf-2f53-417f-b0d2-5d3ea86979f5">

I already tested `Clear project` and run `rm -rf android/app/build android/app/.cxx android/build` but error still happens.

erie-e9 commented:
> Somebody else facing these errors on Android?
> 
> Steps:
> 
> * [ ]  Init new RN project - @0.76.x.
> * [ ]  Install latest version of this dep - yarn add @attarchi/react-native-lottie-splash-screen@2.0.4
> * [ ]  Sync gradle on Android studio.
> * [ ]  Rebuild project: Build output will display next problem.
> 
> ### Error 1
> <img alt="Captura de pantalla 2024-10-31 a la(s) 5 05 46 p m" width="861" src="https://private-user-images.githubusercontent.com/6515342/382137487-5884cf45-1bde-4757-9aae-901c13295a24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwNDk0MTYsIm5iZiI6MTczMjA0OTExNiwicGF0aCI6Ii82NTE1MzQyLzM4MjEzNzQ4Ny01ODg0Y2Y0NS0xYmRlLTQ3NTctOWFhZS05MDFjMTMyOTVhMjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTlUMjA0NTE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzVkMzg2NjEyMTRiOTc4MmM4NmVkYmFmZDJiM2UwMDJmZDljMjc1MTYwYTBkYTliZmQxMjEzYzU0ZmVhZWMyZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.EELbMfARl3m05IHtqbtM5xN_b7TuJa4GGUHjYtQP2bI">
> ### Error 2
> Unresolved reference Lottie in MainActivity.kt <img alt="Captura de pantalla 2024-10-31 a la(s) 5 19 48 p m" width="877" src="https://private-user-images.githubusercontent.com/6515342/382139540-98a6cacf-2f53-417f-b0d2-5d3ea86979f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwNDk0MTYsIm5iZiI6MTczMjA0OTExNiwicGF0aCI6Ii82NTE1MzQyLzM4MjEzOTU0MC05OGE2Y2FjZi0yZjUzLTQxN2YtYjBkMi01ZDNlYTg2OTc5ZjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTlUMjA0NTE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yzc1OTI2NzI5YWRmNmYxNjlmMzI1N2MyZTdkMGI2NjEzYTc5YTQ1YmVjNGMzZmZjNThmZTA3NTEzZmFjNjc2MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.U4AoTbYJ5-eOLACmkm1Wa7YdKxNeEbJYA1yMggEBKRg">
> 
> I already tested `Clear project` and run `rm -rf android/app/build android/app/.cxx android/build` but error still happens.

Solution enable next ones in gradle.properties:
android.useAndroidX=true
android.enableJetifier=true

attarchi commented:
@erie-e9  
I tested this with React Native `0.76.2` and encountered the same error. It seems this issue isn’t directly related to this package; version `0.76.x` might still have some unresolved bugs. Here's how I resolved it:

1. **Yarn vs. npm**  
   I'm using **yarn** instead of **npm**, which might help avoid dependency issues.

2. **Gradle settings**  
   Make sure your `gradle.properties` file includes the following settings:  
   ```gradle
   android.useAndroidX=true
   android.enableJetifier=true
   newArchEnabled=false
   ```
3. Fix for `MainActivity.kt`
You might have missed adding this import to the header of your `MainActivity.kt` file:

```kotlin
import android.os.Bundle
```
4. Ensure correct `lottie-react-native` version
Verify that the `lottie-react-native` package is installed with the exact version `6.7.2`:

```bash
yarn add lottie-react-native@6.7.2
```
5. Please run your project with this command:
```bash
yarn start --reset-cache
```

Let me know if these steps work for you!

erie-e9 commented:
Thank you for your detailed answer. I edited my comment with fix for me, as we mentioned solution is enable those flags. Anyway, I'm using new arch, so for me is working fine without `newArchEnabled=false`

sumeetswn commented:
With enable jetifier flag,
`android.enableJetifier=true
`
I got Max Heap Java Error, which I resolved with increasing the MaxMetaspaceSize from 512M to 1024M.

`org.gradle.jvmargs=-Xmx2048m -XX:MaxMetaspaceSize=1024m
`

erie-e9 commented:
React native v0.77.x is here! Would be good to provide a swift example for this package. I attach my approach to the AppDelegate.swift file:

```
import UIKit
import React
import React_RCTAppDelegate
import ReactAppDependencyProvider

// Third party plugins
import Lottie
import react_native_lottie_splash_screen

@main
class AppDelegate: RCTAppDelegate {
  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
    self.moduleName = "yourAwesomeApp"
    self.dependencyProvider = RCTAppDependencyProvider()

    // You can add your custom initial props in the dictionary below.
    // They will be passed down to the ViewController used by React Native.
    self.initialProps = [:]

    let success = super.application(application, didFinishLaunchingWithOptions: launchOptions)

    if success {
      if let rootView = self.window.rootViewController?.view {
        rootView.backgroundColor = .white

        // Set up Lottie splash screen
        let dynamic = Dynamic()
        if let animationUIView = dynamic.createAnimationView(rootView: rootView, lottieName: "logo_animated") as? UIView {
          react_native_lottie_splash_screen.RNSplashScreen.showLottieSplash(animationUIView, inRootView: rootView)

          if let animationView = animationUIView as? LottieAnimationView {
            dynamic.play(animationView: animationView)
          }

          // Force remove animation layout when hide is called
          react_native_lottie_splash_screen.RNSplashScreen.setAnimationFinished(true)
        }
      }
    }

    return success
  }

  override func sourceURL(for bridge: RCTBridge) -> URL? {
    self.bundleURL()
  }

  override func bundleURL() -> URL? {
#if DEBUG
    RCTBundleURLProvider.sharedSettings().jsBundleURL(forBundleRoot: "index")
#else
    Bundle.main.url(forResource: "main", withExtension: "jsbundle")
#endif
  }
  
}

```


This change has modified some iOS configuration steps for this package. For example, it is no longer necessary to follow Step 2: "Create [your-project-name]-Bridging-Header.h" from the documentation.

Alive!

![Simulator Screen Recording - 16 Pro - 18 2 - 2025-02-14 at 11 51 06](https://github.com/user-attachments/assets/91c2fcf3-a108-4304-ad98-aad0ecb651d0)

attarchi commented:
I’m excited to announce that **version 3** is now complete! This update includes:

* A newly implemented **Swift version** for iOS, alongside various improvements to the setup flows.
* Enhanced setup instructions—refactored to be **easier to follow**, **more robust**, and better aligned with typical development workflows.
* A big **thank you** to everyone for your feedback and contributions that made this possible!

Your input has been invaluable. Looking forward to seeing how these updates help streamline future integrations!

https://github.com/attarchi/react-native-lottie-splash-screen
